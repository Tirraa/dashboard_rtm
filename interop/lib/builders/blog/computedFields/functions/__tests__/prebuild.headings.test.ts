import type { DocumentToCompute } from '@rtm/shared-types/ContentlayerConfig';

import { describe, expect, it } from 'vitest';

import buildBlogPostHeadings from '../headings';

describe('buildBlogPostHeadings', () => {
  it('should match snapshot (empty string)', () => {
    expect(
      buildBlogPostHeadings({
        body: {
          raw: ''
        }
      } as Partial<DocumentToCompute> as DocumentToCompute)
    ).toMatchSnapshot();
  });

  it('should match snapshot (one title)', () => {
    expect(
      buildBlogPostHeadings({
        body: {
          raw: '\n# BAR !'
        }
      } as Partial<DocumentToCompute> as DocumentToCompute)
    ).toMatchSnapshot();
  });

  it('should match snapshot (one title, no heading linefeed)', () => {
    expect(
      buildBlogPostHeadings({
        body: {
          raw: '# BAR !'
        }
      } as Partial<DocumentToCompute> as DocumentToCompute)
    ).toMatchSnapshot();
  });

  it('should match snapshot (not a title)', () => {
    expect(
      buildBlogPostHeadings({
        body: {
          raw: '\n####### BAR !'
        }
      } as Partial<DocumentToCompute> as DocumentToCompute)
    ).toMatchSnapshot();
  });

  it('should match snapshot (not a title, no heading linefeed)', () => {
    expect(
      buildBlogPostHeadings({
        body: {
          raw: '####### BAR !'
        }
      } as Partial<DocumentToCompute> as DocumentToCompute)
    ).toMatchSnapshot();
  });

  it('should match snapshot (melting pot from hell)', () => {
    expect(
      buildBlogPostHeadings({
        body: {
          raw: '\n# BAR !\n\nFOO  \nBAR !\n\n```ts\nfunction foo() {\n  konsole.log(\'BAR\');\n}\n```\n\n## FOO !\n\nFOO !\n\n<Image src="/foo/bar.svg" width="42" height="69" alt="FOO" priority />\n\n### BAR\n\n&nbsp;  \n&nbsp;  \n&nbsp;\n\n#### HAHAHAHA\n\n##### HHHHAAAAAAAAAAAAAAAAAAAA\n\n###### HHHHHHHAAAAAAAAAAAAAAAAA\n\n#### HAHAHAHA\n\n###### HHHHHHHAAAAAAAAAAAAAAAAA\n\n##### HHHHAAAAAAAAAAAAAAAAAAAA\n\n###### HHHHHHHAAAAAAAAAAAAAAAAA\n\n### FOO !\n\n# BAR ### ! ##\n\n## FOO !\n\n####### LOL\n\n# LOL\n\n## LOL\n\n###\n\n## #### LOL\n\n### LOL\n\n#### LOL\n\n##### LOL\n\n###### LOL\n\n####### LOL ##\n\n<InviteTheBotButton />\n'
        }
      } as Partial<DocumentToCompute> as DocumentToCompute)
    ).toMatchSnapshot();
  });

  it('should match snapshot (melting pot from hell, no heading linefeed)', () => {
    expect(
      buildBlogPostHeadings({
        body: {
          raw: '# BAR !\n\nFOO  \nBAR !\n\n```ts\nfunction foo() {\n  konsole.log(\'BAR\');\n}\n```\n\n## FOO !\n\nFOO !\n\n<Image src="/foo/bar.svg" width="42" height="69" alt="FOO" priority />\n\n### BAR\n\n&nbsp;  \n&nbsp;  \n&nbsp;\n\n#### HAHAHAHA\n\n##### HHHHAAAAAAAAAAAAAAAAAAAA\n\n###### HHHHHHHAAAAAAAAAAAAAAAAA\n\n#### HAHAHAHA\n\n###### HHHHHHHAAAAAAAAAAAAAAAAA\n\n##### HHHHAAAAAAAAAAAAAAAAAAAA\n\n###### HHHHHHHAAAAAAAAAAAAAAAAA\n\n### FOO !\n\n# BAR ### ! ##\n\n## FOO !\n\n####### LOL\n\n# LOL\n\n## LOL\n\n###\n\n## #### LOL\n\n### LOL\n\n#### LOL\n\n##### LOL\n\n###### LOL\n\n####### LOL ##\n\n<InviteTheBotButton />\n'
        }
      } as Partial<DocumentToCompute> as DocumentToCompute)
    ).toMatchSnapshot();
  });
});
