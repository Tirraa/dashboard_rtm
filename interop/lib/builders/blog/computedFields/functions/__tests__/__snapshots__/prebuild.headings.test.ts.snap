// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildBlogPostHeadings > should match snapshot (empty string) 1`] = `[]`;

exports[`buildBlogPostHeadings > should match snapshot (melting pot from hell) 1`] = `
[
  {
    "content": "BAR !",
    "depth": 1,
    "slug": "bar-",
  },
  {
    "content": "FOO !",
    "depth": 2,
    "slug": "foo-",
  },
  {
    "content": "BAR",
    "depth": 3,
    "slug": "bar",
  },
  {
    "content": "HAHAHAHA",
    "depth": 4,
    "slug": "hahahaha",
  },
  {
    "content": "HHHHAAAAAAAAAAAAAAAAAAAA",
    "depth": 5,
    "slug": "hhhhaaaaaaaaaaaaaaaaaaaa",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa",
  },
  {
    "content": "HAHAHAHA",
    "depth": 4,
    "slug": "hahahaha-1",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa-1",
  },
  {
    "content": "HHHHAAAAAAAAAAAAAAAAAAAA",
    "depth": 5,
    "slug": "hhhhaaaaaaaaaaaaaaaaaaaa-1",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa-2",
  },
  {
    "content": "FOO !",
    "depth": 3,
    "slug": "foo--1",
  },
  {
    "content": "BAR ### ! ##",
    "depth": 1,
    "slug": "bar---",
  },
  {
    "content": "FOO !",
    "depth": 2,
    "slug": "foo--2",
  },
  {
    "content": "LOL",
    "depth": 1,
    "slug": "lol",
  },
  {
    "content": "LOL",
    "depth": 2,
    "slug": "lol-1",
  },
  {
    "content": "#### LOL",
    "depth": 2,
    "slug": "-lol",
  },
  {
    "content": "LOL",
    "depth": 3,
    "slug": "lol-2",
  },
  {
    "content": "LOL",
    "depth": 4,
    "slug": "lol-3",
  },
  {
    "content": "LOL",
    "depth": 5,
    "slug": "lol-4",
  },
  {
    "content": "LOL",
    "depth": 6,
    "slug": "lol-5",
  },
]
`;

exports[`buildBlogPostHeadings > should match snapshot (melting pot from hell, no heading linefeed) 1`] = `
[
  {
    "content": "BAR !",
    "depth": 1,
    "slug": "bar-",
  },
  {
    "content": "FOO !",
    "depth": 2,
    "slug": "foo-",
  },
  {
    "content": "BAR",
    "depth": 3,
    "slug": "bar",
  },
  {
    "content": "HAHAHAHA",
    "depth": 4,
    "slug": "hahahaha",
  },
  {
    "content": "HHHHAAAAAAAAAAAAAAAAAAAA",
    "depth": 5,
    "slug": "hhhhaaaaaaaaaaaaaaaaaaaa",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa",
  },
  {
    "content": "HAHAHAHA",
    "depth": 4,
    "slug": "hahahaha-1",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa-1",
  },
  {
    "content": "HHHHAAAAAAAAAAAAAAAAAAAA",
    "depth": 5,
    "slug": "hhhhaaaaaaaaaaaaaaaaaaaa-1",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa-2",
  },
  {
    "content": "FOO !",
    "depth": 3,
    "slug": "foo--1",
  },
  {
    "content": "BAR ### ! ##",
    "depth": 1,
    "slug": "bar---",
  },
  {
    "content": "FOO !",
    "depth": 2,
    "slug": "foo--2",
  },
  {
    "content": "LOL",
    "depth": 1,
    "slug": "lol",
  },
  {
    "content": "LOL",
    "depth": 2,
    "slug": "lol-1",
  },
  {
    "content": "#### LOL",
    "depth": 2,
    "slug": "-lol",
  },
  {
    "content": "LOL",
    "depth": 3,
    "slug": "lol-2",
  },
  {
    "content": "LOL",
    "depth": 4,
    "slug": "lol-3",
  },
  {
    "content": "LOL",
    "depth": 5,
    "slug": "lol-4",
  },
  {
    "content": "LOL",
    "depth": 6,
    "slug": "lol-5",
  },
]
`;

exports[`buildBlogPostHeadings > should match snapshot (not a title) 1`] = `[]`;

exports[`buildBlogPostHeadings > should match snapshot (not a title, no heading linefeed) 1`] = `[]`;

exports[`buildBlogPostHeadings > should match snapshot (one title) 1`] = `
[
  {
    "content": "BAR !",
    "depth": 1,
    "slug": "bar-",
  },
]
`;

exports[`buildBlogPostHeadings > should match snapshot (one title, no heading linefeed) 1`] = `
[
  {
    "content": "BAR !",
    "depth": 1,
    "slug": "bar-",
  },
]
`;

exports[`buildBlogPostHeadings should match snapshot (empty string) 1`] = `[]`;

exports[`buildBlogPostHeadings should match snapshot (one title) 1`] = `
[
  {
    "content": "BAR !",
    "depth": 1,
    "slug": "bar-",
  },
]
`;

exports[`buildBlogPostHeadings should match snapshot (one title, no heading linefeed) 1`] = `
[
  {
    "content": "BAR !",
    "depth": 1,
    "slug": "bar-",
  },
]
`;

exports[`buildBlogPostHeadings should match snapshot (not a title) 1`] = `[]`;

exports[`buildBlogPostHeadings should match snapshot (not a title, no heading linefeed) 1`] = `[]`;

exports[`buildBlogPostHeadings should match snapshot (melting pot from hell) 1`] = `
[
  {
    "content": "BAR !",
    "depth": 1,
    "slug": "bar-",
  },
  {
    "content": "FOO !",
    "depth": 2,
    "slug": "foo-",
  },
  {
    "content": "BAR",
    "depth": 3,
    "slug": "bar",
  },
  {
    "content": "HAHAHAHA",
    "depth": 4,
    "slug": "hahahaha",
  },
  {
    "content": "HHHHAAAAAAAAAAAAAAAAAAAA",
    "depth": 5,
    "slug": "hhhhaaaaaaaaaaaaaaaaaaaa",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa",
  },
  {
    "content": "HAHAHAHA",
    "depth": 4,
    "slug": "hahahaha-1",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa-1",
  },
  {
    "content": "HHHHAAAAAAAAAAAAAAAAAAAA",
    "depth": 5,
    "slug": "hhhhaaaaaaaaaaaaaaaaaaaa-1",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa-2",
  },
  {
    "content": "FOO !",
    "depth": 3,
    "slug": "foo--1",
  },
  {
    "content": "BAR ### ! ##",
    "depth": 1,
    "slug": "bar---",
  },
  {
    "content": "FOO !",
    "depth": 2,
    "slug": "foo--2",
  },
  {
    "content": "LOL",
    "depth": 1,
    "slug": "lol",
  },
  {
    "content": "LOL",
    "depth": 2,
    "slug": "lol-1",
  },
  {
    "content": "#### LOL",
    "depth": 2,
    "slug": "-lol",
  },
  {
    "content": "LOL",
    "depth": 3,
    "slug": "lol-2",
  },
  {
    "content": "LOL",
    "depth": 4,
    "slug": "lol-3",
  },
  {
    "content": "LOL",
    "depth": 5,
    "slug": "lol-4",
  },
  {
    "content": "LOL",
    "depth": 6,
    "slug": "lol-5",
  },
]
`;

exports[`buildBlogPostHeadings should match snapshot (melting pot from hell, no heading linefeed) 1`] = `
[
  {
    "content": "BAR !",
    "depth": 1,
    "slug": "bar-",
  },
  {
    "content": "FOO !",
    "depth": 2,
    "slug": "foo-",
  },
  {
    "content": "BAR",
    "depth": 3,
    "slug": "bar",
  },
  {
    "content": "HAHAHAHA",
    "depth": 4,
    "slug": "hahahaha",
  },
  {
    "content": "HHHHAAAAAAAAAAAAAAAAAAAA",
    "depth": 5,
    "slug": "hhhhaaaaaaaaaaaaaaaaaaaa",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa",
  },
  {
    "content": "HAHAHAHA",
    "depth": 4,
    "slug": "hahahaha-1",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa-1",
  },
  {
    "content": "HHHHAAAAAAAAAAAAAAAAAAAA",
    "depth": 5,
    "slug": "hhhhaaaaaaaaaaaaaaaaaaaa-1",
  },
  {
    "content": "HHHHHHHAAAAAAAAAAAAAAAAA",
    "depth": 6,
    "slug": "hhhhhhhaaaaaaaaaaaaaaaaa-2",
  },
  {
    "content": "FOO !",
    "depth": 3,
    "slug": "foo--1",
  },
  {
    "content": "BAR ### ! ##",
    "depth": 1,
    "slug": "bar---",
  },
  {
    "content": "FOO !",
    "depth": 2,
    "slug": "foo--2",
  },
  {
    "content": "LOL",
    "depth": 1,
    "slug": "lol",
  },
  {
    "content": "LOL",
    "depth": 2,
    "slug": "lol-1",
  },
  {
    "content": "#### LOL",
    "depth": 2,
    "slug": "-lol",
  },
  {
    "content": "LOL",
    "depth": 3,
    "slug": "lol-2",
  },
  {
    "content": "LOL",
    "depth": 4,
    "slug": "lol-3",
  },
  {
    "content": "LOL",
    "depth": 5,
    "slug": "lol-4",
  },
  {
    "content": "LOL",
    "depth": 6,
    "slug": "lol-5",
  },
]
`;
